core/common/assets/js/api/modules/component-modal-base.js

```diff
@@ -14,6 +14,11 @@ export default class ComponentModalBase extends ComponentBase {
 	defaultCommands() {
 		const self = this;
 
+		/**
+		 * Base component cannot have commands under folder because:
+		 * `CommandBase` know only which component attached to command while it runs it.
+		 * Here it uses 'self'.
+		 */
```

Comment:
You should pass it as an argument to the command

Reply:
to pass the component as arg?

Reply:
Instead of using `self` and closure to identify the active command, pass the `ComponentModalBase` instance as an argument to the command class, then you'll be able to put each class in a different file

Reply:
We can't touch the command base for this edge case.

Reply:
Handled!